rest_lights.cpp

```diff
@@ -1327,30 +1330,61 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
 
     bool requestOk = true;
     bool hasCmd = false;
-    bool hasOn = false;
-    bool targetOn = false;
+    bool hasOpen = false;
+    bool targetOpen = false;
     bool hasLift = false;
-    bool hasBri = false;
     bool hasStop = false;
-    quint8 targetLiftPct = 0;
+    quint8 targetLift = 0;
+    quint8 targetLiftZigBee = 0;
     bool hasTilt = false;
-    quint8 targetTiltPct = 0;
+    quint8 targetTilt = 0;
 
     // Check parameters.
     for (QVariantMap::const_iterator p = map.begin(); p != map.end(); p++)
     {
         bool paramOk = false;
         bool valueOk = false;
         QString param = p.key();
-        if (param == "on" && taskRef.lightNode->item(RStateOn))
+        if (param == "open" && taskRef.lightNode->item(RStateOpen))
         {
             paramOk = true;
             hasCmd = true;
-            if (map["on"].type() == QVariant::Bool)
+            if (map[param].type() == QVariant::Bool)
             {
                 valueOk = true;
-                hasOn = true;
-                targetOn = map["on"].toBool();
+                hasOpen = true;
+                targetOpen = map[param].toBool();
+            }
+        }
+        else if (param == "on" && taskRef.lightNode->item(RStateOn))
+        {
+            paramOk = true;
+            hasCmd = true;
+            if (map[param].type() == QVariant::Bool)
+            {
+                valueOk = true;
+                hasOpen = true;
+                targetOpen = !(map[param].toBool());

```

Comment:
The open attribute is nice, so if I understand correctly, for clients which still use the "on" attribute nothing will break?

Reply:
That's the idea.  Both `on` and `open` are reported by GET.  You can still PUT `on` but the API reports `open` instead of `on` in the response, to ease people into using the `open`.  Same for `bri` vs `lift` and `sat` vs `tilt`.

Reply:
Perfect, really like the more meaningful attribute names here.